tests/setup-permissions.test.sh

```
@@ -0,0 +1,123 @@
+
+# 5d. .env is not group/other writable and not world-readable (0640)
+m="$(mode "$SB/.env")"
+[ "$m" = "640" ] && ok ".env is 0640 (secret-safe)" || bad ".env mode is $m (want 640)"
+
+# 5e. data dir is group-writable (0775)
+m="$(mode "$SB/storage")"
+[ "$m" = "775" ] || [ "$m" = "2775" ] && ok "storage is group-writable ($m)" || bad "storage mode is $m (want 775/2775)"
+
+# 5f. NOTHING is world-writable (no 777, no o+w anywhere)
+_ww="$(find "$SB" -perm -0002 2>/dev/null | grep -v '/storage/sessions' | head -1)"
+[ -z "$_ww" ] && ok "nothing is world-writable (no 777)" || bad "world-writable path found: $_ww"
```

Shellcheck (0.11.0)

[info] Note that A && B || C is not if-then-else. C may run when A is true.

(SC2015)